src/main/kotlin/com/netflix/dgs/plugin/hints/DgsInputArgumentInspector.kt

```diff
@@ -0,0 +1,237 @@
+        return if (isJavaFile) {
+            getHintForInputArgumentInJava(input)
+        } else {
+            getHintForInputArgumentInKotlin(input)
+        }
+    }
+
+    private fun getHintForInputArgumentInJava(input: GraphQLInputValueDefinition) : String? {
+        val argName = (input.nameIdentifier as GraphQLIdentifierImpl).name
+        var inputArgumentHint = "@InputArgument "
+        if (input.type is GraphQLListType) {
+            val collectionType = getCollectionType(input.type, true)
+            if (! isSimpleType(collectionType)) {
+                inputArgumentHint = "$inputArgumentHint(collectionType=$collectionType.class) "
+            }
+        }
+        inputArgumentHint = inputArgumentHint + getType(input.type, true)  + " " + argName
+        return inputArgumentHint
+    }
+
+    private fun getHintForInputArgumentInKotlin(input: GraphQLInputValueDefinition) : String? {
+        val argName = (input.nameIdentifier as GraphQLIdentifierImpl).name
+        var inputArgumentHint = "@InputArgument "
+        if (input.type is GraphQLListType) {
+            val collectionType = getCollectionType(input.type, false)
+            if (! isSimpleType(collectionType)) {
+                inputArgumentHint = "$inputArgumentHint(collectionType=$collectionType) "
+            }
+        }
+        inputArgumentHint = inputArgumentHint + argName + ": "+ getType(input.type, false)  + " "
+        return inputArgumentHint
+    }
+
+    private fun isSimpleType(typeName: String?) : Boolean {
+        return typeName.equals("String") || typeName.equals("Integer") || typeName.equals("Int") || typeName.equals("Boolean") || typeName.equals("Double")
+    }
+
+    private fun getType(inputType: GraphQLType?, isJavaType: Boolean) : String? {
+        return when (inputType) {
+            is GraphQLTypeName -> {
+                val rawType = getRawType((inputType as GraphQLTypeNameImpl).name, isJavaType)
+                if (isJavaType) {
+                    rawType
+                } else {
+                    "$rawType?"
+                }
+            }
+            is GraphQLListType -> {
+                val type = "List<" + getType(inputType.type, isJavaType) + ">"
+                if (isJavaType) {
+                    type
+                } else {
+                    "$type?"
+                }
+            }
+            is GraphQLNonNullType -> {
+                val type = getType(inputType.type, isJavaType)
+                if (isJavaType) {
+                    type
+                } else {
+                    type?.removeSuffixIfPresent("?")
+                }
+            }
+            else -> ""
+        }
+    }
+
+    private fun getCollectionType(inputType: GraphQLType?, isJavaFile: Boolean) : String? {
+        return when (inputType) {
+            is GraphQLTypeName -> {
+                getRawType((inputType as GraphQLTypeNameImpl).name, isJavaFile)
```

Comment:
Is the cast to `GraphQLTypeNameImpl` necessary? Isn’t there something like a `Named` in the type hierarchy?

Reply:
Switched to PsiNamedElement